### DISCUSSION TITLE
[solved] Login after existin checkout doesnt assign order to customer correctly

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#11358] [General] [answered]
Hi guys,
I've got an issue with login and cart (order when completed).

If I add something to the cart and after I create an account and I login completing the order, this order doesn't appear in my history.  
A workaround is create a new checkout after I logged in adn assign the old cart lines to the new checkout.

There's no errors then is not useful past any graphql response. 
In any case (and in my case) https://docs.saleor.io/docs/3.x/developer/users#previous-orders doesn't work properly.

Do you have notice this behaviour?
It's not a problem use the workaround but these are unnecessary  abandoned cart.

Thanks in advance and kind regards,

Davide

**Comment by karolkielecki** [comment]
Hi @DidoMarchet, Could you please tell us which version you are using?

**Comment by korycins** [comment, marked as answer]
@DidoMarchet  Not sure if I fully understand. 
Do you mean, you create anonymous checkout, then log in and finalize the checkout?
In that case, did you call `CheckoutCustomerAttach` mutation to attach the anonymous checkout to the user?

**Reply by DidoMarchet** [reply]
Hi @karolkielecki and @korycins yes! I did't find it! When you have time I think could be useful add it into the doc where checkout or user behavior are explained!
Thanks for the support and kind regards, 
Davide

**Reply by korycins** [reply]
Hey @DidoMarchet 
Thanks for the reply. You're right, the explanation of these mutations is missing. I've created the issue to track this. https://github.com/saleor/saleor-docs/issues/608
